core/vdbe/mod.rs

```diff
@@ -1754,6 +1759,20 @@ fn exec_substring(
     }
 }
 
+fn exec_typeof(reg: &OwnedValue) -> OwnedValue {
+    match reg {
+        OwnedValue::Null => OwnedValue::Text(Rc::new("null".to_string())),
+        OwnedValue::Integer(_) => OwnedValue::Text(Rc::new("integer".to_string())),
+        OwnedValue::Float(_) => OwnedValue::Text(Rc::new("real".to_string())),
+        OwnedValue::Text(_) => OwnedValue::Text(Rc::new("text".to_string())),
+        OwnedValue::Blob(_) => OwnedValue::Text(Rc::new("blob".to_string())),
+
+        // TODO: Unreachable?
+        OwnedValue::Record(_) => todo!(),
+        OwnedValue::Agg(_) => todo!(),
```

Comment:
According to SQLite [`typeof` documentation](https://www.sqlite.org/lang_corefunc.html#typeof) the return value is either "null", "integer", "real", "text", or "blob". Maybe we should use `unreachable!()` macro here and believe that we never run function with this type of inputs.